azurerm/internal/services/synapse/synapse_sql_pool_resource.go

```diff
@@ -0,0 +1,378 @@
+	}
+
+	transparentDataEncryption, err := sqlPTDEClient.Get(ctx, id.Workspace.ResourceGroup, id.Workspace.Name, id.Name)
+	if err != nil {
+		return fmt.Errorf("retrieving Transparent Data Encryption settings of Synapse SqlPool %q (Resource Group %q / workspaceName %q): %+v", id.Name, id.Workspace.ResourceGroup, id.Workspace.Name, err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("synapse_workspace_id", id.Workspace.String())
+	if resp.Sku != nil {
+		d.Set("sku_name", resp.Sku.Name)
+	}
+	if props := resp.SQLPoolResourceProperties; props != nil {
+		d.Set("collation", props.Collation)
+	}
+	if props := transparentDataEncryption.TransparentDataEncryptionProperties; props != nil {
+		d.Set("data_encrypted", props.Status == synapse.TransparentDataEncryptionStatusEnabled)
+	}
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceArmSynapseSqlPoolDelete(d *schema.ResourceData, meta interface{}) error {
+	sqlClient := meta.(*clients.Client).Synapse.SqlPoolClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.SynapseSqlPoolID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	future, err := sqlClient.Delete(ctx, id.Workspace.ResourceGroup, id.Workspace.Name, id.Name)
+	if err != nil {
+		return fmt.Errorf("deleting Synapse SqlPool %q (Resource Group %q / workspaceName %q): %+v", id.Name, id.Workspace.ResourceGroup, id.Workspace.Name, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, sqlClient.Client); err != nil {
+		return fmt.Errorf("waiting on deleting future for Synapse SqlPool %q (Resource Group %q / workspaceName %q): %+v", id.Name, id.Workspace.ResourceGroup, id.Workspace.Name, err)
+	}
+	return nil
+}
+
+func synapseSqlPoolScaleStateRefreshFunc(ctx context.Context, client *synapse.SQLPoolsClient, resourceGroup, workspaceName, name string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		resp, err := client.Get(ctx, resourceGroup, workspaceName, name)
+		if err != nil {
+			return resp, "failed", err
+		}
+		if resp.SQLPoolResourceProperties == nil || resp.SQLPoolResourceProperties.Status == nil {
+			return resp, "failed", nil
+		}
+		return resp, *resp.SQLPoolResourceProperties.Status, nil
+	}
+}
+
+// sqlPool backend service is a proxy to sql database
+// backend service restore and backup only accept id format of sql database
+// so if the id is sqlPool, we need to construct the corresponding sql database id
+func constructSourceDatabaseId(id string) string {
+	sqlPoolId, err := parse.SynapseSqlPoolID(id)
+	if err != nil {
+		return id
+	}
+	sqlDataBaseId := sqlParse.SqlDatabaseId{
+		SubscriptionID: sqlPoolId.Workspace.SubscriptionID,
+		ResourceGroup:  sqlPoolId.Workspace.ResourceGroup,
+		ServerName:     sqlPoolId.Workspace.Name,
+		Name:           sqlPoolId.Name,
+	}
+	return sqlDataBaseId.String()
```

Comment:
rather than doing this - can we switch to using [the ResourceIDFormatter approach](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/frontdoor/parse/health_probe.go#L5-L22) which [ensures they comply with the interface](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/frontdoor/parse/health_probe_test.go#L9)